### PR TITLE
Add cartographic to cartesian3 method

### PR DESCRIPTION
**Pull request by hanbollar**
oneliner method to shorten conversion from cartographic to cartesian3.

inside the method just calls fromRadians - instead of needing to input the cartographic values into the fromRadians specifically

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@hanbollar, thanks for the pull request! Maintainers, we have a signed CLA from @hanbollar, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by ggetz** [comment]
Thanks @hanbollar! Please update `CHANGES.md` and add a test.

**Comment by mramato** [comment]
Please don't merge this until we take a closer look at the implications.  While it appears like a small change, it actually creates a circular dependency and the built version of cesium will most likely have issues.  `Cartographic` requires in `scaleToGeodeticSurface`, which requires in `Cartesian3`, which requires in `Cartographic`.  There's a reason we never added this function.

**Comment by lilleyse** [comment]
For reference, this was the PR that added `Cartographic.fromCartesian`: https://github.com/AnalyticalGraphicsInc/cesium/pull/3193/files

Maybe the same ideas apply here?

**Comment by hanbollar** [comment]
@mramato Thanks for the feedback! I swapped the location of this conversion method from the Cartesian3 file to the Cartographic file to bypass the circular include issue.

File: Source/Core/Cartographic.js [hunk 1]
```diff
@@ -146,6 +148,24 @@ define([
...
+     * Returns a Cartesian3 position from a {@link Cartographic} input.
+     *
+     * @param {Cartographic} Cartographic input to be converted into a Cartesian3 output.
+     * @param {Ellipsoid} [ellipsoid=Ellipsoid.WGS84] The ellipsoid on which the position lies.
+     * @param {Cartesian3} [result] The object onto which to store the result.
+     * @returns {Cartesian3} The position
+     */
+    Cartographic.toCartesian3 = function(cartographic, ellipsoid, result) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(cartographic)) {
```
Comment (mramato) [inline Source/Core/Cartographic.js]: This block can be replaced with `Check.defined('cartographic', cartographic)`, which is out new way of doing parameter checking.

File: Source/Core/Cartographic.js [hunk 2]
```diff
@@ -146,6 +148,24 @@ define([
         return result;
     };
 
+    /**
+     * Returns a Cartesian3 position from a {@link Cartographic} input.
+     *
+     * @param {Cartographic} Cartographic input to be converted into a Cartesian3 output.
```
Comment (mramato) [inline Source/Core/Cartographic.js]: Lowercase c.

**Comment by hanbollar** [comment]
@mramato Updated following your comments.

File: Specs/Core/CartographicSpec.js [hunk 3]
```diff
@@ -27,6 +27,16 @@ defineSuite([
         expect(c.height).toEqual(3);
     });
 
+    it('toCartesian3', function(){
```
Comment (ggetz) [inline Specs/Core/CartographicSpec.js]: explain what your test is supposed to do: `toCartesian3 converts a Cartographic to a Cartesian3`

File: Source/Core/Cartographic.js [hunk 4]
```diff
@@ -146,6 +146,22 @@ define([
         return result;
     };
 
+    /**
+     * Returns a Cartesian3 position from a {@link Cartographic} input.
```
Comment (ggetz) [inline Source/Core/Cartographic.js]: `Creates a new {@link Cartesian3} instance from a Cartographic.`

No need to link to Cartographic, we're already in that class.

**Comment by hanbollar** [comment]
@ggetz Updated with your recommended changes.

File: Source/Core/Cartographic.js [hunk 5]
```diff
@@ -146,6 +146,22 @@ define([
...
 
+    /**
+     * Returns a Cartesian3 position from a Cartographic input.
+     *
+     * @param {Cartographic} cartographic Input to be converted into a Cartesian3 output.
+     * @param {Ellipsoid} [ellipsoid=Ellipsoid.WGS84] The ellipsoid on which the position lies.
+     * @param {Cartesian3} [result] The object onto which to store the result.
+     * @returns {Cartesian3} The position
+     */
+    Cartographic.toCartesian3 = function(cartographic, ellipsoid, result) {
```
Comment (ggetz) [inline Source/Core/Cartographic.js]: One more thing, let's be consistent with `fromCartesian` and call this `toCartesian`. Also keep the wording consistent with the other function, see https://github.com/AnalyticalGraphicsInc/cesium/pull/6163/files#diff-f2664d42919c57ae2aa0b2575097bb56R54 and  https://github.com/AnalyticalGraphicsInc/cesium/pull/6163/files#diff-f2664d42919c57ae2aa0b2575097bb56R82

**Comment by hanbollar** [comment]
@ggetz Thanks for the feedback - added commits to match previous conventions better

**Comment by ggetz** [comment]
Thanks @hanbollar, I'll merge when the build passes.

**Pull request merged**